File: Java/1302. Deepest Leaves Sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        if (root == null) {
            return 0;
        }
        
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        int sum = 0;
        int size;
        
        while (!q.isEmpty()) {
            size = q.size();
            sum = 0;
            
            for (int i=0; i<size; i++) {
                root = q.poll();
                sum += root.val;
                
                if (root.left != null) {
                    q.offer(root.left);
                }
                
                if (root.right != null) {
                    q.offer(root.right);
                }
            }
        }
        
        return sum;
    }
}

File: Python3/1302. Deepest Leaves Sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        q = deque([root])
        while q:
            level = len(q)
            _sum = 0
            
            for _ in range(level):
                root = q.popleft()
                _sum += root.val
                
                if root.left:
                    q.append(root.left)
                    
                if root.right:
                    q.append(root.right)
        
        return _sum